bin/evm2near/src/compile.rs

```diff
@@ -111,24 +140,33 @@ impl Compiler {
             abi_buffer_off: find_abi_buffer(&runtime_library).unwrap(),
             abi_buffer_len: 0xFFFF, // TODO: ensure this matches _abi_buffer.len() in evmlib
             op_table: make_op_table(&runtime_library),
-            jump_table: HashMap::new(),
-            function_type: find_runtime_function_type(&runtime_library).unwrap(),
             evm_start_function: 0, // filled in during emit_start()
             evm_init_function: find_runtime_function(&runtime_library, "_evm_init").unwrap(),
             evm_call_function: find_runtime_function(&runtime_library, "_evm_call").unwrap(),
             evm_post_exec_function: find_runtime_function(&runtime_library, "_evm_post_exec")
                 .unwrap(),
             evm_exec_function: 0, // filled in during compile_cfg()
             evm_pop_function: find_runtime_function(&runtime_library, "_evm_pop_u32").unwrap(),
+            evm_push_function: find_runtime_function(&runtime_library, "_evm_push_u32").unwrap(),
+            evm_burn_gas: find_runtime_function(&runtime_library, "_evm_burn_gas").unwrap(),
             evm_pc_function: find_runtime_function(&runtime_library, "_evm_set_pc").unwrap(),
             function_import_count: runtime_library.import_count(ImportCountType::Function),
             builder: parity_wasm::builder::from_module(runtime_library),
         }
     }
 
+    fn debug<TPath: Into<PathBuf>, CF: Fn() -> String>(&self, path: TPath, contents: CF) {
+        if let Some(base_path) = &self.config.debug_path {
+            let mut full_path = base_path.clone();
+            full_path.push(path.into());
+
+            std::fs::write(full_path, contents()).expect("fs error while writing debug file");
+        }
+    }
+
     /// Emit an empty `_start` function to make all WebAssembly runtimes happy.
     fn emit_wasm_start(self: &mut Compiler) {
-        _ = self.emit_function(Some("_start".to_string()), vec![]);
+        _ = self.emit_function(Some("_start".to_string()), vec![Instruction::Unreachable]);
```

Comment:
Tests shows that it's work, but @shishkin-pavel can you explain it more detailly

Reply:
unlike `_start`function in C-like languages (runtime initialization routine), `_start` in wasm seems to be just a default entry point
so, wasmtime invocation without entry point specification will start that function and return "succesfully", which is probably not the thing we meant to do anyways, thats why I thought that getting process terminated with error is better than silently do nothing and pass

Reply:
According to the [wasm specification](https://www.w3.org/TR/wasm-core-1/#start-function%E2%91%A0)

> The [start](https://www.w3.org/TR/wasm-core-1/#syntax-module) component of a module declares the [function index](https://www.w3.org/TR/wasm-core-1/#syntax-funcidx) of a start function that is automatically invoked when the module is [instantiated](https://www.w3.org/TR/wasm-core-1/#exec-instantiation), after [tables](https://www.w3.org/TR/wasm-core-1/#syntax-table) and [memories](https://www.w3.org/TR/wasm-core-1/#syntax-mem) have been initialized.

And you can see [this happening](https://docs.rs/wasmer-vm-near/2.4.0/src/wasmer_vm/instance/mod.rs.html#958-972) in the Wasm library the Near Runtime uses.

Since running contracts works without hitting the `unreachable` I think it must mean we are not actually assigning `_start` as being the official start function of the wasm module. Perhaps this is something to look into.

Reply:
oh, thanks! its quite strange that wasmtime doesnt call that function either